src/dask_awkward/lib/structure.py

```diff
@@ -467,22 +654,97 @@ def pad_none(
 
 @borrow_docstring(ak.ravel)
 def ravel(array, highlevel=True, behavior=None):
-    raise DaskAwkwardNotImplemented("TODO")
+    if not highlevel:
+        raise ValueError("Only highlevel=True is supported")
+
+    if isinstance(array._meta.layout, ak.contents.recordarray.RecordArray):
+        warnings.warn("ravel may produce inconsistent results for record arrays!")
```

Comment:
What do you mean by inconsistent results here. Should we just not allows it if it can't be consistent?

Reply:
They are correct results but out of order row-wise comapared to awkward. So not wrong but possibly confusing if validating against raw awkward.

Reply:
OK that seems like a reasonable thing we can field if a question comes up